Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -345,49 +368,55 @@ extension AWSClient {
     /// - returns:
     ///     A signed URL
     public func signURL(url: URL, httpMethod: String, expires: Int = 86400) -> EventLoopFuture<URL> {
-        return signer.map { signer in signer.signURL(url: url, method: HTTPMethod(rawValue: httpMethod), expires: expires) }
```

Comment:
Shouldn't move this to service configuration as you need a signer, which you access from the client. Would be odd to have one code path ie execute go via client and another signURL go via ServiceConfig
```
public func signURL(url: URL, httpMethod: String, configuration: ServiceConfig, expires: Int)
```
I am debating whether to add S3 helper functions in aws-sdk-swift as S3.GetObject and S3.PutObject are the most common uses of this.